checker/src/org/checkerframework/checker/optional/OptionalChecker.java

```diff
@@ -0,0 +1,22 @@
+package org.checkerframework.checker.optional;
+
+import java.util.LinkedHashSet;
+import org.checkerframework.checker.nullness.NullnessChecker;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+
+/**
+ * A type-checker that prevents misuse of the {@link java.util.Optional} class.
+ *
+ * @checker_framework.manual #optional-checker Optional Checker
+ */
+// Uncomment when the Checker Framework assumes Java 8: @RelevantJavaTypes(Optional.class)
+public class OptionalChecker extends BaseTypeChecker {
+
+    @Override
+    protected LinkedHashSet<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
```

Comment:
I think the Optional Checker would be much better without the dependency on the Nullness Checker, as described in previous review comments.
This will only have an impact on `Optional.of` and `Optional.ofNullable` calls, which we can leave as future issues.
The Optional Checker would then be a fast checker for that specific property. Handling the interaction with the Nullness Checker can then be done in another iteration.
The impact to this PR should be small: remove this method, the uses of nullnessTypeFactory, and move a few tests to a new issue.